be/src/exec/pipeline/sort/local_partition_topn_source.cpp

```diff
@@ -17,7 +17,12 @@ bool LocalPartitionTopnSourceOperator::has_output() const {
 }
 
 bool LocalPartitionTopnSourceOperator::is_finished() const {
-    return _partition_topn_ctx->is_finished();
+    return _is_finished || _partition_topn_ctx->is_finished();
```

Comment:
Seems that nobody calls `_partition_topn_ctx->set_finished`.
Should call `ctx->set_finished` in `SourceOperator::set_finished` and call `ctx->is_finished` in `SinkOperator::is_finished`?